csrc/host_ir/jit.cpp

```diff
@@ -387,62 +619,79 @@ HostIrJit::HostIrJit(
       mangler,
       kTensorSizeFuncName);
   registerExternalFunction(
-      allocate_tensor_func_ptr,
-      name_to_symbol,
-      mangler,
-      kAllocateTensorFuncName);
+      new_tensor_func_ptr, name_to_symbol, mangler, kNewTensorFuncName);
+  registerExternalFunction(
+      delete_tensor_func_ptr, name_to_symbol, mangler, kDeleteTensorFuncName);
   registerExternalFunction(
       set_tensor_func_ptr, name_to_symbol, mangler, kSetTensorFuncName);
+  registerExternalFunction(
+      empty_strided_cuda_func_ptr,
+      name_to_symbol,
+      mangler,
+      kAtEmptyStridedCudaWrapper);
   throwIfError(
       dest_dynamic_lib.define(llvm::orc::absoluteSymbols(name_to_symbol)));
-  // Compile the module
-  compile(pimpl_.get());
 }
 
-KernelArgumentHolder HostIrJit::runWithInputs(
+KernelArgumentHolder HostIrJitImpl::runWithInputs(
     const KernelArgumentHolder& args) {
-  FUSER_PERF_SCOPE("HostIrJit::runWithInputs");
+  FUSER_PERF_SCOPE("HostIrJitImpl::runWithInputs");
   // Bind cache id to llvm global variable or align with main function inputs
   NVF_ERROR(args.getCacheId().has_value(), "Cache ID is not set");
-  NVF_ERROR_EQ(std::ssize(pimpl_->container->inputs()), args.size());
+  NVF_ERROR_EQ(std::ssize(container_->inputs()), args.size());
 
   std::vector<const void*> input_aten_tensors;
   // Bind the inputs to the tensor map
-  for (auto&& [in_val, arg] : zip(pimpl_->container->inputs(), args)) {
+  for (auto&& [in_val, arg] : zip(container_->inputs(), args)) {
```

Comment:
```suggestion
  for (auto [in_val, arg] : zip(container_->inputs(), args)) {
```
I probably suggested `auto&&` initially but realized it's an overkill in this case. int and pointer are cheap to copy by value so `for (auto` or `for (const auto` is perfectly fine.